src/components/Connections/ConnectTesla.tsx

```diff
@@ -33,14 +41,15 @@ export const ConnectTesla: FC = () => {
 
     if (!stateFromUrl) return;
 
+    const state = JSON.parse(stateFromUrl);
```

Comment:
```suggestion
    const extratedJsonState = JSON.parse(stateFromUrl);
```
or something more expressive than `state`